server/server.go

```diff
@@ -124,24 +154,38 @@ func (s *Server) newConn(conn net.Conn) *clientConn {
 		}
 	}
 	cc.setConn(conn)
-	cc.salt = util.RandomBuf(20)
+	cc.salt = util.RandomBuf(mysql.ScrambleLength)
 	return cc
 }
 
+func (s *Server) newXConn(conn net.Conn) *mysqlXClientConn {
+	return &mysqlXClientConn{
+		conn:         conn,
+		pkt:          xpacketio.NewXPacketIO(conn),
+		server:       s,
+		capability:   defaultCapability,
+		connectionID: atomic.AddUint32(&baseConnID, 1),
+		collation:    mysql.DefaultCollationID,
+		alloc:        arena.NewAllocator(32 * 1024),
+		salt:         util.RandomBuf(mysql.ScrambleLength),
+	}
+}
+
 func (s *Server) skipAuth() bool {
 	return s.cfg.Socket != ""
 }
 
 const tokenLimit = 1000
 
 // NewServer creates a new Server.
-func NewServer(cfg *config.Config, driver IDriver) (*Server, error) {
+func NewServer(cfg *config.Config, driver IDriver, serverType int) (*Server, error) {
 	s := &Server{
 		cfg:               cfg,
 		driver:            driver,
 		concurrentLimiter: NewTokenLimiter(tokenLimit),
 		rwlock:            &sync.RWMutex{},
-		clients:           make(map[uint32]*clientConn),
+		tp:                serverType,
```

Comment:
is this protocolType instead of serverType?

Reply:
Done.